src/test/java/co/rsk/federate/config/SignerConfigBuilder.java

```diff
@@ -0,0 +1,54 @@
+package co.rsk.federate.config;
+
+import static com.typesafe.config.ConfigValueFactory.fromAnyRef;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import co.rsk.federate.signing.config.SignerConfig;
+
+public class SignerConfigBuilder {
+  private Config config;
+
+  private SignerConfigBuilder() {
+    this.config = ConfigFactory.empty();
+  }
+
+  public static SignerConfigBuilder builder() {
+    return new SignerConfigBuilder();
+  }
+
+  public SignerConfigBuilder withValue(String path, Object value) {
+    this.config = this.config.withValue(path, fromAnyRef(value));
+    return this;
+  }
+
+  public SignerConfigBuilder withHsmSigner(String keyId) {
+    this.config = this.config.withValue("type", fromAnyRef("hsm"));
+    this.config = this.config.withValue("host", fromAnyRef("127.0.0.1"));
+    this.config = this.config.withValue("port", fromAnyRef(9999));
+    this.config = this.config.withValue("keyId", fromAnyRef(keyId));
+    this.config = this.config.withValue("socketTimeout", fromAnyRef(20000));
+    this.config = this.config.withValue("maxAttempts", fromAnyRef(3));
+    this.config = this.config.withValue("intervalBetweenAttempts", fromAnyRef(2000));
+    return this;
+  }
+
+  public SignerConfigBuilder withHsmBookkeepingInfo() {
```

Comment:
```suggestion
  public SignerConfigBuilder withHsmBookkeepingInfo(BigInteger difficultyTarget, long informerInterval ....) {
```
This is what I had in mind actually.

That way it can called simply `withHsmBookkeepingInfo(target, interval, ...` and there's no need to know the name of the fields that need to be set.

Makes sense? 

If you think it makes sense to also have some default values, perhaps we could set those when calling `withHsmSigner`. We can have a quick chat and settle on this

Reply:
Okay I see, I have updated the method with this approach. I also allowed it for the `difficultyTarget` to be nullable so we can use it for the test case where `difficultyTarget` is not set in the config. Let me know if that is fine.